crates/vm/src/system/memory/adapter/mod.rs

```diff
@@ -75,6 +75,13 @@ impl<F> AccessAdapterInventory<F> {
         debug_assert!(chip.n() == n);
```

Comment:
perf related: can we have `AccessAdapterRecord` store the `log2_len` so we don't need to do this `log2_strict_usize` above?

not related to this PR

Reply:
I guess unless we could do some compile time this with const generic (seems unlikely), it's not possible

Reply:
Hmm, this will just move the `log2_strict_usize` to the `AccessAdapterRecord` construction rather than here, so I'm not sure it gains anything?

We could make `chips` an `FxHashMap` keyed by `n`, but I'm not sure `hash(n)` will be much faster than `log2_strict_usize`. We can use `n.trailing_zeros()` instead of `log2_strict_usize` though, since we don't need the assertion that `n` is a power of two (that should have been checked earlier).